### PR TITLE
Issue fixes of triangle count mismatch and docker mode

### PR DESCRIPTION
**Pull request by chinthakarukshan**

### CODE REVIEW

File: src/server/JasmineGraphInstanceService.cpp [hunk 1]
```diff
@@ -640,7 +640,32 @@ void *instanceservicesession(void *dummyPt) {
...
+                std::string chunk = aggregatedTriangles.substr(i, INSTANCE_DATA_LENGTH-10);
+                if (i + INSTANCE_DATA_LENGTH-10 < aggregatedTriangles.length()) {
+                    chunk += "/SEND";
+                } else {
+                    chunk += "/CMPT";
+                }
+                chunksVector.push_back(chunk);
+            }
+
+            for (int loopCount=0;loopCount < chunksVector.size();loopCount++) {
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:656]: Add spaces between variables. Lets change as follows,

```
 for (int loopCount = 0; loopCount < chunksVector.size(); loopCount++) {
```

File: src/server/JasmineGraphInstanceService.cpp [hunk 2]
```diff
@@ -640,7 +640,32 @@ void *instanceservicesession(void *dummyPt) {
 
 
             std::string aggregatedTriangles= JasmineGraphInstanceService::aggregateCentralStoreTriangles(graphId, partitionId, partitionIdList);
-            write(connFd, aggregatedTriangles.c_str(), aggregatedTriangles.size());
+
+            std::vector<std::string> chunksVector;
+
+            for (unsigned i = 0; i < aggregatedTriangles.length(); i += INSTANCE_DATA_LENGTH-10) {
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:646]: Add space between variables. Change as follows,

```
INSTANCE_DATA_LENGTH - 10
```

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 3]
```diff
@@ -1675,6 +1675,24 @@ string JasmineGraphFrontEnd::countCentralStoreTriangles(std::string aggregatorHo
             read(sockfd, data, 300);
             response = (data);
             response = utils.trim_copy(response, " \f\n\r\t\v");
+            string status = response.substr(response.size()-5);
```
Comment (miyurud) [inline src/frontend/JasmineGraphFrontEnd.cpp:1678]: Add space between variables. Change as,

```
response.substr(response.size() - 5);
```

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 4]
```diff
@@ -1675,6 +1675,24 @@ string JasmineGraphFrontEnd::countCentralStoreTriangles(std::string aggregatorHo
...
+                result_wr = write(sockfd, status.c_str(), status.size());
+
+                if(result_wr < 0) {
+                    frontend_logger.log("Error writing to socket", "error");
+                }
+                bzero(data, 301);
+                read(sockfd, data, 300);
+                response = (data);
+                response = utils.trim_copy(response, " \f\n\r\t\v");
+                status = response.substr(response.size()-5);
```
Comment (miyurud) [inline src/frontend/JasmineGraphFrontEnd.cpp:1691]: Add space between variables

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 5]
```diff
@@ -1675,6 +1675,24 @@ string JasmineGraphFrontEnd::countCentralStoreTriangles(std::string aggregatorHo
...
+                std::string triangleResponse= response.substr(0, response.size()-5);
```
Comment (miyurud) [inline src/frontend/JasmineGraphFrontEnd.cpp:1692]: Add space between variables

File: src/server/JasmineGraphInstanceService.cpp [hunk 6]
```diff
@@ -640,7 +640,32 @@ void *instanceservicesession(void *dummyPt) {
...
+                std::string chunk = aggregatedTriangles.substr(i, INSTANCE_DATA_LENGTH-10);
+                if (i + INSTANCE_DATA_LENGTH-10 < aggregatedTriangles.length()) {
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:648]: Add space between variables

File: src/server/JasmineGraphInstanceService.cpp [hunk 7]
```diff
@@ -640,7 +640,32 @@ void *instanceservicesession(void *dummyPt) {
 
 
             std::string aggregatedTriangles= JasmineGraphInstanceService::aggregateCentralStoreTriangles(graphId, partitionId, partitionIdList);
-            write(connFd, aggregatedTriangles.c_str(), aggregatedTriangles.size());
+
+            std::vector<std::string> chunksVector;
+
+            for (unsigned i = 0; i < aggregatedTriangles.length(); i += INSTANCE_DATA_LENGTH-10) {
+                std::string chunk = aggregatedTriangles.substr(i, INSTANCE_DATA_LENGTH-10);
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:647]: Add space between variables

**Review by miyurud — CHANGES_REQUESTED** [review]
Please address the comments

File: src/server/JasmineGraphInstanceService.cpp (hunk 1 above)
Comment (chinthakarukshan) [reply src/server/JasmineGraphInstanceService.cpp]: Updated the code

File: src/frontend/JasmineGraphFrontEnd.cpp (hunk 3 above)
Comment (chinthakarukshan) [reply src/frontend/JasmineGraphFrontEnd.cpp]: Updated the code

File: src/server/JasmineGraphInstanceService.cpp (hunk 6 above)
Comment (chinthakarukshan) [reply src/server/JasmineGraphInstanceService.cpp]: Updated the code

File: src/server/JasmineGraphInstanceService.cpp (hunk 7 above)
Comment (chinthakarukshan) [reply src/server/JasmineGraphInstanceService.cpp]: Updated the code

File: src/server/JasmineGraphInstanceService.cpp (hunk 6 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

**Pull request merged**